testing/inference/score_function_test.py

```diff
@@ -0,0 +1,125 @@
+import unittest
+import mxfusion as mf
+import mxnet as mx
+import numpy as np
+import mxnet.gluon.nn as nn
+from mxfusion.components.variables.var_trans import PositiveTransformation
+from mxfusion.components.functions import MXFusionGluonFunction
+from mxfusion.util.testutils import make_basic_model
+
+
+class ScoreFunctionTests(unittest.TestCase):
```

Comment:
We need a test case for the correctness of the inference method, maybe compare with the gradient from the ones from reparameterization tricks.